### ISSUE TITLE
Using "Long Syntax" for volumes generates `PHP Warning:  explode() expects parameter 2 to be string, array given`

### ISSUE THREAD

**Issue by karlwilbur** [pmsipilot/docker-compose-viz#39]
Running from the CLI, I get:
```
-> % ~/packages/docker-compose-viz/bin/dcv render
PHP Warning:  explode() expects parameter 2 to be string, array given in /home/karl/packages/docker-compose-viz/src/functions.php on line 504
PHP Notice:  Undefined offset: 0 in /home/karl/packages/docker-compose-viz/src/functions.php on line 513
PHP Notice:  Undefined offset: 0 in /home/karl/packages/docker-compose-viz/src/functions.php on line 513
PHP Warning:  explode() expects parameter 2 to be string, array given in /home/karl/packages/docker-compose-viz/src/functions.php on line 504
PHP Notice:  Undefined offset: 0 in /home/karl/packages/docker-compose-viz/src/functions.php on line 513
PHP Notice:  Undefined offset: 0 in /home/karl/packages/docker-compose-viz/src/functions.php on line 513
```

Running from the Docker image also gives similar warnings:
```
-> % docker run --rm -it --name dcv -v $(pwd):/input pmsipilot/docker-compose-viz render -m image docker-compose.yml

Warning: explode() expects parameter 2 to be string, array given in /dcv/src/functions.php on line 504

Warning: explode() expects parameter 2 to be string, array given in /dcv/src/functions.php on line 504
```

In my `docker-compose.yml` I am using the [long syntax](https://docs.docker.com/compose/compose-file/#long-syntax-3); I have the following:
```
    volumes:
      - type: bind
        source: ./app-source
        target: /place/to/bin/app-source
      - type: bind
        source: ./logs
        target: /place/to/bin/logs
```
...which is what seems to be causing the trouble when read in by this line:
https://github.com/pmsipilot/docker-compose-viz/blob/78c83037ee8903f6b6da14c987ad37b4466206e2/src/functions.php#L309

It seems that this only supports the [short syntax](https://docs.docker.com/compose/compose-file/#short-syntax-3). 

I am going to see if I can put together a fix for this and submit a PR.

**Comment by Xenira** [comment]
Is this related to #36?

**Comment by jonmitten** [comment]
Same issue here. Would love this to become patched.

**Comment by karlwilbur** [comment]
@Xenira, Yes, I believe this is related to #36.

@jonmitten, I made a fix for this and submitted a PR (#40). It was approved but has not yet been merged.

**Comment by karlwilbur** [comment]
@Xenira, I just looked at the PR you submitted (#37) and it is identical to mine (#40). I'm not sure why your is failing CI checks.
